pkg/apis/v1alpha1/common.go

```diff
@@ -191,3 +192,39 @@ type FieldSelectorRequirement struct {
 	// the values array must be empty.
 	Values []string `json:"values,omitempty"`
 }
+
+type RealizedResource struct {
+	// Name is the name of the resource in the blueprint
+	Name string `json:"name"`
+
+	// StampedRef is a reference to the object that was created by the resource
+	StampedRef *corev1.ObjectReference `json:"stampedRef,omitempty"`
+
+	// TemplateRef is a reference to the template used to create the object in StampedRef
+	TemplateRef *corev1.ObjectReference `json:"templateRef,omitempty"`
+
+	// Inputs are references to resources that were used to template the object in StampedRef
+	Inputs []Input `json:"inputs,omitempty"`
+
+	// Outputs are values from the object in StampedRef that can be consumed by other resources
+	Outputs []Output `json:"outputs,omitempty"`
+}
+
+type Input struct {
+	// Name is the name of the resource in the blueprint whose output the resource consumes as an input
+	Name string `json:"name"`
+}
+
+type Output struct {
+	// Name is the output type generated from the resource [url, revision, image or config]
+	Name string `json:"name"`
+
+	// Preview is a preview of the value of the output
```

Comment:
given that the value one finds here is a string whose contents are json-encoded,I think we should call that out here in the comments for anyone willing to interpret that somehow (ala `JSON.parse()` in javascript ..)

Reply:
since it is a preview, we could be cutting it off and then you would not be able to interpret with JSON.parse

Reply:
uh yes, that's a good point indeed 👍